paymentsheet/src/main/java/com/stripe/android/paymentsheet/forms/TransformSpecToElement.kt

```diff
@@ -51,21 +47,17 @@ private fun FormItemSpec.SectionSpec.transform(
     )
 }
 
-private fun List<SectionFieldSpec>.transform(
-    focusRequesterCount: FocusRequesterCount
-) = this.map {
+private fun List<SectionFieldSpec>.transform() = this.map {
     when (it) {
-        is SectionFieldSpec.Email -> it.transform(focusRequesterCount)
-        is SectionFieldSpec.Iban -> it.transform(focusRequesterCount)
+        is SectionFieldSpec.Email -> it.transform()
+        is SectionFieldSpec.Iban -> it.transform()
         is SectionFieldSpec.Country -> it.transform()
         is SectionFieldSpec.IdealBank -> it.transform()
-        is SectionFieldSpec.SimpleText -> it.transform(focusRequesterCount)
+        is SectionFieldSpec.SimpleText -> it.transform()
```

Comment:
Looks like this `when` is no longer necessary. Can `transform` be a fun in `SectionFieldSpec`?

Reply:
The cast is provided by the when is needed as transform is an extension function on the specific spec.   

I like having the transformations happen outside of the specific classes, as spec is a concrete class, and one that is intended to be the json sent down from the server.